python/ambassador/ir/irhost.py

```diff
@@ -391,17 +394,17 @@ def save_context(self, ir: "IR", ctx_name: str, tls_ss: SavedSecret, tls_name: s
         # TLS config is good, let's make sure the hosts line up too.
         context_hosts = ctx.get("hosts")
 
-        # XXX WTF? self.name is not OK as a hostname! Leaving this for the moment, but it's
-        # almost certainly getting shredded before 2.0 GAs.
-        host_hosts = [self.name]
```

Comment:
as we discussed offline, I dug into this deeper and although we don't document this anywhere there is a potential that a user could ignore our docs and intpret the `TLSContext.hosts` as needing to match the name of a `Host.name`.

Although, it is unlikely and it is not our intended behavior,I will add it back since it is not crucial to the fix for this PR. I will capture it along with the other deprecated TLS features that I outlined and we can plan for its removal in the future.

Reply:
Here is the overall issue on this topic:
https://github.com/emissary-ingress/emissary/issues/4826

I make a mention to this and that it is deprecated. I will have a fixup commit here shortly that does remove some dead code where we check `self.hostname or self.name` because we default self.hostname to "*" if user doesn't provide it.

Reply:
FTR I'm the one who wrote that comment. `self.name` is not guaranteed to be something that's actually sane for a DNS match, IIRC, so I'm in favor of it going away. However, it used to be important that `host_hosts` never be empty; might be worth a test.

Reply:
(Newer folks, if you see a comment that starts with `XXX`, it's almost certainly something I wrote. 🙂)